.github/workflows/docker_build_and_push.yaml

```diff
@@ -0,0 +1,38 @@
+name: Docker Build and Push
+permissions:
+  checks: write
+  contents: write
+  pull-requests: write
+  statuses: write
+  id-token: write
+"on":
+  push:
+jobs:
+  build_push_image:
+    runs-on: ubuntu-latest
+    needs: test_and_lint
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - uses: launchdarkly/common-actions/init@main
+      - uses: launchdarkly/common-actions/short-sha@main
+        id: short-sha
+      - uses: launchdarkly/common-actions/aws-auth@main
```

Comment:
Commenting for posterity as we already discussed, but you'll need to use the primitive actions here, rather than our internal ones as they're not public